File: tools/lua-bindings/autobuild/huawei.lua
-- AUTO BUILD, DON'T MODIFY!

dofile "autobuild/huawei-types.lua"

name = "huawei"
path = "../../frameworks/plugins/huawei"
headers = [[
    #include "lua-bindings/lua_conv.h"
    #include "lua-bindings/lua_conv_manual.h"
    #include "huawei/huawei.h"
]]
chunk = nil
luaopen = nil


typeconf 'cclua::plugin::huawei'
    .supercls(nil)
    .reg_luatype(true)
    .chunk(nil)
    .luaopen(nil)
    .indexerror(nil)
    .ifdef('*', '#ifdef CCLUA_BUILD_HUAWEI')
    .func(nil, 'static void init()')
    .func(nil, 'static void revokeAccess()')
    .func(nil, 'static void logout()')
    .func(nil, 'static void login()')
    .func(nil, 'static void pay(const std::string &appid, const std::string &cpid, const std::string &cp, const std::string &orderNo, const std::string &pid, const std::string &url, const std::string &serviceCatalog, const std::string &extReserved, const std::string &sign)')
    .func(nil, 'static void dispatch(const std::string &event, const cocos2d::ValueMap &data)')
    .callback {
        funcs =  {
            'static void setDispatcher(@localvar const cclua::Callback &dispatcher)'
        },
        tag_maker = 'Dispatcher',
        tag_mode = 'replace',
        tag_store = 0,
        tag_scope = 'object',
    }

File: frameworks/plugins/huawei/huawei.cpp
#include "huawei.h"

#if defined(CCLUA_BUILD_HUAWEI) && defined(CCLUA_OS_ANDROID)

USING_NS_CCLUA;
USING_NS_CCLUA_PLUGIN;

const char *huawei::JAVA_CLASS = "cclua/plugin/huawei/Huawei";
Callback huawei::_dispatcher;

void huawei::init()
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "init");
}

void huawei::revokeAccess()
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "revokeAccess");
}

void huawei::logout()
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "logout");
}

void huawei::login()
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "login");
}

void huawei::pay(const std::string &appid, const std::string &cpid, const std::string &cp,
        const std::string &orderNo, const std::string &pid, const std::string &url,
        const std::string &serviceCatalog, const std::string &extReserved, const std::string &sign)
{
    Jni::callStaticVoidMethod(JAVA_CLASS, "pay", appid, cpid, cp, orderNo, pid, url, serviceCatalog,
        extReserved, sign);
}

#endif